core/commands/partials/terraform_cleaner.rb

```diff
@@ -41,7 +43,11 @@ def destroy_nodes(nodes, path, provider, configuration_id)
 
     @ui.info('Destroying the machines using terraform')
     result = TerraformService.resource_type(provider).and_then do |resource_type|
-      resources = TerraformService.nodes_to_resources(nodes, resource_type).values
+      if resource_type == "ibm_pi_instance"
+        resources = TerraformService.ibm_nodes_to_resources(nodes)
```

Comment:
Please pin extra dependencies in generated terraform file with correct depends_on signature: https://developer.hashicorp.com/terraform/tutorials/configuration-language/dependencies